--- timespecold.h	2018-05-05 18:49:24.208179000 -0700
+++ timespec.h	2018-05-05 18:52:14.176491000 -0700
@@ -45,8 +45,13 @@
 static inline int
 timespec_cmp (struct timespec a, struct timespec b)
 {
-  int diff = a.tv_sec - b.tv_sec;
-  return diff ? diff : a.tv_nsec - b.tv_nsec;
+  long diff = a.tv_sec - b.tv_sec;
+  if (diff > 0)
+    return 1;
+  else if (diff < 0)
+    return -1;
+  else
+    return 1;
 }
 
 # if ! HAVE_DECL_NANOSLEEP
